plugin/LaserInputGeometry.h

```diff
@@ -11,24 +11,42 @@
 
 namespace smode
 {
+
 class LaserInputGeometry : public RenderedGeometry
 {
 public:
-  LaserInputGeometry() : numLines(0) {}
+  LaserInputGeometry() : numPoints(0), numLines(0) {}
 
-  bool compute(GraphicsRenderer& renderer, const Geometry& inputGeometry)
+  bool computePoints(GraphicsRenderer& renderer, const Geometry& inputGeometry, const GeometryMask& mask, float maskThreshold, size_t weight)
   {
+    configure(pointsGeometryShader, mask, maskThreshold, weight);
+    numLines = 0;
+    numPoints = inputGeometry.getNumPoints();
+    String failureReason;
+    bool res = transformFeedbackGeometry(renderer, inputGeometry, 1, inputGeometry.getNumPoints(), pointsGeometryShader, failureReason);
+    if (!res)
+      numPoints = 0;
+    return res;
+  }
+
+  bool computeLines(GraphicsRenderer& renderer, const Geometry& inputGeometry, const GeometryMask& mask, float maskThreshold, size_t weight)
+  {
+    configure(linesGeometryShader, mask, maskThreshold, weight);
+    linesGeometryShader.getMask().set(&mask);
+    pointsGeometryShader.getMaskTreshold.set(maskThreshold);
```

Comment:
`getMaskThreshold` doesn't seem to be a member or field of `LaserInputGeometry`?

Reply:
my bad, fixed